driver/main.c

```diff
@@ -1523,13 +1523,14 @@ static inline int drop_event(struct ppm_consumer_t *consumer,
 	}
 
 	if (consumer->dropping_mode) {
+		nanoseconds ns2 = ns;
 		if (drop_flags & UF_ALWAYS_DROP) {
 			ASSERT((drop_flags & UF_NEVER_DROP) == 0);
 			return 1;
 		}
 
 		if (consumer->sampling_interval < SECOND_IN_NS &&
-		    (ns % SECOND_IN_NS) >= consumer->sampling_interval) {
+		    do_div(ns2, SECOND_IN_NS) >= consumer->sampling_interval) {
```

Comment:
The behavior of do_div is very much not obvious (in that it returns the remainder and modifies the dividend to contain the division result) so it could use a comment I think.

Reply:
added a comment